README.md

```diff
@@ -83,9 +85,20 @@ The kube-prometheus stack includes a resource metrics API server, so the metrics
 $ minikube addons disable metrics-server
 ```
 
+## Compatibility
+
+### Kubernetes compatibility matrix
+
+| kube-prometheus stack | Kubernetes 1.14 | Kubernetes 1.15 | Kubernetes 1.16 | Kubernetes 1.17 |
+|-----------------------|-----------------|-----------------|-----------------|-----------------|
+| `release-0.2`         | ✔              | ✔              | ✗              | ✗              |
+| `release-0.3`         | ✔              | ✔              | ✔              | ✔              |
```

Comment:
Various cAdvisor labels were completely dropped in 1.16, I can't imagine we're compatible at this entire range for `release-0.3`.

Reply:
What do you suggest then?

Reply:
release-0.3 not being compatible with 1.16 and 1.17, or only 1.16 and 1.17, I'm not exactly sure which changes are in release-0.3, but it's not possible for us to be compatible with all these versions because of the breaking changes that happened with metrics labels

Reply:
cAdvisor dropped labels were only used in the `experimental` directory so there is no impact on `release-0.3` compatibility with 1.16 and 1.17

Reply:
Just a note: we started using `release-0.3` in OpenShift 4.3, which roughly translates to kubernetes 1.16. Are we sure this branch works in k8s pre-1.16 (apart from cAdvisor changes)?

Reply:
I verified the compatibility by doing the "quickstart" on branch release-0.3 inside both a 1.14 and a 1.15 cluster. In both cases I checked that:

- creation went well
- all resources are running
- no worrisome alerts triggered

After verifying everything I concluded that release-0.3 was compatible with 1.14 and 1.15.

Reply:
>no worrisome alerts triggered

Were there some alerts triggering? When you say 1.14 and 1.15 you mean vanilla k8s?

Reply:
> Were there some alerts triggering? 

Yes the following ones:
- KubeCPUOvercommit
- KubeMemOvercommit
- KubeControllerManagerDown
- KubeSchedulerDown

But since they seemed to trigger every time, I felt that they were not problematic

> When you say 1.14 and 1.15 you mean vanilla k8s?

Yes, I was running the minikube command given in the README and just changed the `kubernetes-version` argument value to match the version I was targetting.

Reply:
> Various cAdvisor labels were completely dropped in 1.16, I can't imagine we're compatible at this entire range for release-0.3.

https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.16.md#removed-metrics
As far as I know, it only affects "container_name" and "pod_name" metrics.

After running a few greps it appears that:
- In 0.1 these metrics were used everywhere (dashboard ...)
- In 0.2 only in experimental and prometheus-adapter
- In 0.3/0.4 only in experimental

@brancz from what I understand, it should not affect 0.3 compatibility

Reply:
Ah right, we had the migration period, so starting 1.14 the `pod` and `container` labels are available, so this is correct.

Reply:
> Yes the following ones:
KubeCPUOvercommit
KubeMemOvercommit
KubeControllerManagerDown
KubeSchedulerDown

Yes, that's normal, that is because you are testing on minikube cluster, right?

Reply:
Can you check one last thing, can you look at the logs of kube-state-metrics when you launch a 1.17 cluster with the release-0.3 branch. I suspect there might be some errors there.

Reply:
> Yes, that's normal, that is because you are testing on minikube cluster, right?

Yes

Reply:
Side note, not related to this PR: https://kubernetes.io/docs/setup/learning-environment/minikube/#configuring-kubernetes There are some docs for minikube on how to configure controller-manager on minikube k8s cluster, if you are interested.